### PR TITLE
GATEWAY OPTIMISATION - MOB SPAWNING

### PR DESCRIPTION
**Pull request by Gandalf2k15**
## About The Pull Request

Mobs no longer instantly spawn on mob_spawn_initialise. They instead spawn when a mob WITH a client enters their trigger range.

This should help with performance a little.

## Changelog

:cl:
code: Some mob spawners have been optimised.
/:cl:

### CODE REVIEW

File: modular_skyrat/master_files/code/modules/awaymission/mission_code/black_mesa/mesa_mobs.dm [hunk 1]
```diff
@@ -549,6 +549,8 @@
 	desc = "A 5.56mm surplus bullet casing."
 	projectile_type = /obj/projectile/bullet/a556/weak
 
+#define MOB_PLACER_RANGE 12 // One more tile than the biggest viewrange we have.
```
Comment (nevimer) [inline modular_skyrat/master_files/code/modules/awaymission/mission_code/black_mesa/mesa_mobs.dm]: CONFIG_GET(string/default_view) + 1 ?

Comment (Gandalf2k15) [reply modular_skyrat/master_files/code/modules/awaymission/mission_code/black_mesa/mesa_mobs.dm]: That relies on the config being set tho

Comment (GoldenAlpharex) [reply modular_skyrat/master_files/code/modules/awaymission/mission_code/black_mesa/mesa_mobs.dm]: I don't see many instances where the config wouldn't be set.

File: modular_skyrat/master_files/code/modules/awaymission/mission_code/black_mesa/mesa_mobs.dm [hunk 2]
```diff
@@ -549,6 +549,8 @@
 	desc = "A 5.56mm surplus bullet casing."
 	projectile_type = /obj/projectile/bullet/a556/weak
 
+#define MOB_PLACER_RANGE CONFIG_GET(string/default_view) + 1 // One more tile than the biggest viewrange we have.
```
Comment (GoldenAlpharex) [inline modular_skyrat/master_files/code/modules/awaymission/mission_code/black_mesa/mesa_mobs.dm]: Actually I think that config might not give us the thing we actually want, as that seems to be a string that says `15x15`.

I'd probably say something like 8 would be a good value here, and if other servers have for some reason an even bigger distance, well... Sucks to be them I guess.

Unless we can find the define that's already saying the size of our widescreen.

File: modular_skyrat/master_files/code/modules/awaymission/mission_code/black_mesa/mesa_mobs.dm [hunk 3]
```diff
@@ -557,9 +559,22 @@
 
 /obj/effect/random_mob_placer/Initialize(mapload)
 	. = ..()
+	for(var/turf/iterating_turf in range(MOB_PLACER_RANGE, src))
+		RegisterSignal(iterating_turf, COMSIG_ATOM_ENTERED, .proc/trigger)
```
Comment (GoldenAlpharex) [inline modular_skyrat/master_files/code/modules/awaymission/mission_code/black_mesa/mesa_mobs.dm:563]: What happens if those turfs are deleted?

File: modular_skyrat/master_files/code/modules/awaymission/mission_code/black_mesa/mesa_mobs.dm [hunk 4]
```diff
@@ -557,9 +559,22 @@
...
+
+/obj/effect/random_mob_placer/proc/trigger(datum/source, atom/movable/entered_atom)
+	if(!isliving(entered_atom))
```
Comment (GoldenAlpharex) [inline modular_skyrat/master_files/code/modules/awaymission/mission_code/black_mesa/mesa_mobs.dm:567]: You should probably add `SIGNAL_HANDLER` on the first line of the proc, as that's typically what's done for procs that are activated by a signal, if I remember correctly?

**Review by GoldenAlpharex — CHANGES_REQUESTED** [review]

File: modular_skyrat/master_files/code/modules/awaymission/mission_code/black_mesa/mesa_mobs.dm (hunk 3 above)
Comment (Gandalf2k15) [reply modular_skyrat/master_files/code/modules/awaymission/mission_code/black_mesa/mesa_mobs.dm:563]: They just lose their signal register

File: modular_skyrat/master_files/code/modules/awaymission/mission_code/black_mesa/mesa_mobs.dm (hunk 4 above)
Comment (Gandalf2k15) [reply modular_skyrat/master_files/code/modules/awaymission/mission_code/black_mesa/mesa_mobs.dm:567]: Yee

File: modular_skyrat/master_files/code/modules/awaymission/mission_code/black_mesa/mesa_mobs.dm (hunk 2 above)
Comment (nevimer) [reply modular_skyrat/master_files/code/modules/awaymission/mission_code/black_mesa/mesa_mobs.dm]: The default *should be* 21x15, and the non default should be 15x15, as a fresh server defaults to 21x15.

File: modular_skyrat/master_files/code/modules/awaymission/mission_code/black_mesa/mesa_mobs.dm [hunk 5]
```diff
@@ -549,7 +549,7 @@
 	desc = "A 5.56mm surplus bullet casing."
 	projectile_type = /obj/projectile/bullet/a556/weak
 
-#define MOB_PLACER_RANGE CONFIG_GET(string/default_view) + 1 // One more tile than the biggest viewrange we have.
+#define MOB_PLACER_RANGE 16 // One more tile than the biggest viewrange we have.
```
Comment (nevimer) [inline modular_skyrat/master_files/code/modules/awaymission/mission_code/black_mesa/mesa_mobs.dm:552]: I dunno, I see issues down the line if TG ever added viewport adjustment or some of the downstreams edit their default config.

**Pull request merged**